code/go/0chain.net/smartcontract/storagesc/writepool.go

```diff
@@ -1,489 +1,145 @@
-		wp = new(writePool)
-	}
-
-	bps, err := makeCopyAllocationBlobbers(*alloc, txn.Value)
-	if err != nil {
-		return fmt.Errorf("error creating blobber pools: %v", err)
-	}
-	var ap = allocationPool{
-		AllocationID: alloc.ID,
-		ExpireAt:     alloc.Until(),
-		Blobbers:     bps,
-	}
-	ap.TokenPool.ID = txn.Hash
-	alloc.addWritePoolOwner(alloc.Owner)
-	wp.Pools.add(&ap)
-
-	if err = wp.save(ssc.ID, alloc.Owner, balances); err != nil {
-		return fmt.Errorf("saving write pool: %v", err)
-	}
-
-	return
-}
-
-func (ssc *StorageSmartContract) createWritePool(
-	t *transaction.Transaction,
-	alloc *StorageAllocation,
-	mintNewTokens bool,
-	balances chainState.StateContextI,
-) (err error) {
-	var wp *writePool
-	wp, err = ssc.getWritePool(alloc.Owner, balances)
-
-	if err != nil && err != util.ErrValueNotPresent {
-		return fmt.Errorf("getting client write pool: %v", err)
-	}
-
-	if err == util.ErrValueNotPresent {
-		wp = new(writePool)
-	}
-
-	var mld = alloc.restMinLockDemand()
-	if t.Value < mld || t.Value == 0 {
-		return fmt.Errorf("not enough tokens to honor the min lock demand"+
-			" (%d < %d)", t.Value, mld)
-	}
-
-	if t.Value > 0 {
-		var until = alloc.Until()
-		ap, err := newAllocationPool(t, alloc, until, mintNewTokens, balances)
-		if err != nil {
-			return err
-		}
-		wp.Pools.add(ap)
-	}
-
-	if err = wp.save(ssc.ID, alloc.Owner, balances); err != nil {
-		return fmt.Errorf("saving write pool: %v", err)
-	}
-
-	return
-}
-
-// lock tokens for write pool of transaction's client
-func (ssc *StorageSmartContract) writePoolLock(t *transaction.Transaction,
-	input []byte, balances chainState.StateContextI) (resp string, err error) {
-
-	var conf *writePoolConfig
-	if conf, err = ssc.getWritePoolConfig(balances, true); err != nil {
+	input []byte,
+	balances cstate.StateContextI,
+) (string, error) {
+	var conf *Config
+	var err error
+	if conf, err = ssc.getConfig(balances, true); err != nil {
 		return "", common.NewError("write_pool_lock_failed",
 			"can't get configs: "+err.Error())
 	}
 
-	// lock request & user balance
-
 	var lr lockRequest
 	if err = lr.decode(input); err != nil {
 		return "", common.NewError("write_pool_lock_failed", err.Error())
 	}
-
-	if len(lr.TargetId) == 0 {
-		lr.TargetId = t.ClientID
-	}
-
-	var wp *writePool
-	if wp, err = ssc.getWritePool(lr.TargetId, balances); err != nil {
-		if err != util.ErrValueNotPresent {
-			return "", common.NewError("write_pool_lock_failed", err.Error())
-		}
-		wp = new(writePool)
-	}
+	logging.Logger.Info("piers writePoolLock", zap.Any("input", lr))
 
 	if lr.AllocationID == "" {
 		return "", common.NewError("write_pool_lock_failed",
 			"missing allocation ID in request")
 	}
 
-	if t.Value < conf.MinLock || t.Value == 0 {
-		return "", common.NewError("write_pool_lock_failed",
-			"insufficient amount to lock")
+	iTxnVal := txn.Value
+	if err != nil {
+		return "", err
 	}
-
-	if lr.Duration < conf.MinLockPeriod {
+	if iTxnVal < conf.WritePool.MinLock || txn.Value <= 0 {
```

Comment:
is it possible that txn.Value < 0?

Reply:
Ok so txn.Value has been changed to unit, so it can't be any longer.